File: backend/app/schemas/team.py
"""Team schema definitions."""
from __future__ import annotations

from typing import Dict, List, Optional

from pydantic import BaseModel


class Team(BaseModel):
    id: int
    name: str
    short_name: str
    city: Optional[str] = None  # Changed from home_venue to city
    home_venue: Optional[str] = None  # Keep for backward compatibility
    founded_year: Optional[int] = None

    class Config:
        orm_mode = True


class TeamDetail(Team):
    squad_size: int
    squad_value: float
    avg_age: float
    international_players: int
    role_distribution: Dict[str, int]


class TeamComparison(BaseModel):
    teams: List[TeamDetail]
    comparison: Dict[str, List[float]]

File: frontend/src/types/team.ts
export interface Team {
  id: number
  name: string
  short_name: string
  home_venue?: string
  founded_year?: number
}

export interface TeamDetail extends Team {
  squad_size: number
  squad_value: number
  avg_age: number
  international_players: number
  role_distribution: Record<string, number>
}

export interface TeamComparison {
  teams: TeamDetail[]
  comparison: {
    avg_squad_value: number[]
    avg_age: number[]
    international_players: number[]
  }
}
